osversion/osversion.go

```diff
@@ -0,0 +1,67 @@
+package osversion
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// OSVersion is a wrapper for Windows version information
+// https://msdn.microsoft.com/en-us/library/windows/desktop/ms724439(v=vs.85).aspx
+type OSVersion struct {
+	Version      uint32
+	MajorVersion uint8
+	MinorVersion uint8
+	Build        uint16
+}
+
+func newVersion(majorVersion, minorVersion uint8, buildNumber uint16) OSVersion {
+	osv := OSVersion{
+		MajorVersion: majorVersion,
+		MinorVersion: minorVersion,
+		Build:        buildNumber,
+	}
+	// Fill version value so that existing clients don't break
+	osv.Version = uint32(buildNumber) << 16
+	osv.Version |= uint32(osv.MinorVersion) << 8
+	osv.Version |= uint32(osv.MajorVersion)
+	return osv
+}
+
+// Parse parses a string representation of OSVersion.
```

Comment:
_perhaps_ we could add an example input value to outline the expected format (if there's a reference link somewhere, I guess that could be useful, but not strictly necessary).